src/icrar/leap-accelerate/ms/utils.h

```diff
@@ -106,12 +130,15 @@ namespace icrar
         casacore::Slicer array_section(start2, length2);
 
         // Read the data.
-        casacore::ArrayColumn<std::complex<double>> ac(ms, column);
-        casacore::Array<std::complex<double>> column_range = ac.getColumnRange(row_range, array_section);
+        casacore::ArrayColumn<std::complex<T>> ac(ms, column);
+        std::cout << row_range << std::endl;
+        std::cout << array_section << std::endl;
+        casacore::Array<std::complex<T>> column_range = ac.getColumnRange(row_range, array_section);
```

Comment:
This code from oskar currently fails in the tests with the following:
```
1: [0] to [195] with stride [1], length [196]
1: [0, 0] to [2, 99] with stride [1, 1], length [3, 100]
1: unknown file: Failure
1: C++ exception with description "Slicer error:infer: endResult>=shape" thrown in the test body.
1: [  FAILED  ] MSUtilsTests.test_read_vis (331 ms)
```

Currently investigating the circumstances for when this happens.

Reply:
The number of polarizations (3) seems weird, it's usually 2 or 4. If your MS has 2 polarizations then it would explain the problem, but if it has 4 I guess it should be fine.

Reply:
I updated the test to read the value from the file. 4 seems to work. I also realized this c styled function assumes the complex output layout is `T real, imag`. I'll need to verify if I need to swizzle the data to form a `std::vector<MVuvw>`

Reply:
`std::complex<T>` is guaranteed to have the same layout as `T[2]`, with the real part in the first element, so I think you'll be safe there.